### PR TITLE
implemented conversion of bayer to Mono, RGB1, RGB2 and RGB3

### PR DESCRIPTION
**Pull request by arvinderpalaha**
For issue #399 

Implemented a very simple linear interpolation for creating rgb values from a bayer colour filter array image, ignoring the pixels on the edges of the image.

This has been tested at DLS on a RHEL7 machine/ioc and it converts bayer to mono/rgb successfully, with a corresponding drop in network usage.

### CODE REVIEW

**Comment by MarkRivers** [comment]
Do you have any performance measurements for this?

**Comment by arvinderpalaha** [comment]
Hi Mark, I don't have any performance measurements for this.

If you'd like me to, I can go about taking some measurements. Is there a preferred way of doing these sorts of tests?

**Comment by MarkRivers** [comment]
You can run your detector in Bayer mode with this plugin converting to RGB1.  In the plugin screen it will show the execution time in ms.  That should be enough to see what the performance is.

**Comment by arvinderpalaha** [comment]
I've seen for some other plugins that performance measurements were taken using ADSimDetector. I could try using that (after adding a bayer color mode to that as well :) )

**Comment by arvinderpalaha** [comment]
I did as you suggested, and got the following on a rhel6 workstation with a 3.6 GHz 4C/8T Intel W-2123 cpu and 16 GB of ram:

Input | Output | SizeX | Size Y | Average (ms) | s.d. (ms) | samples
-- | -- | -- | -- | -- | -- | --
rgb | rgb | 1292 | 964 | 1.436 | 0.344 | 103
rgb | rgb | 646 | 482 | 0.488 | 0.142 | 158
bayer | rgb | 1292 | 964 | 19.534 | 3.247 | 154
bayer | rgb | 646 | 482 | 9.306 | 2.912 | 148
bayer | mono | 1292 | 964 | 20.889 | 2.737 | 151
bayer | mono | 646 | 482 | 8.985 | 2.927 | 135

File: ADApp/pluginSrc/NDPluginColorConvert.cpp [hunk 1]
```diff
@@ -38,12 +38,14 @@ void NDPluginColorConvert::convertColor(NDArray *pArray)
...
     epicsType *pIn, *pRedIn, *pGreenIn, *pBlueIn;
     epicsType *pOut, *pRedOut, *pGreenOut, *pBlueOut;
+    epicsType *p1, *p2, *p3, *p4, *p6, *p7, *p8, *p9; // pointers used for bayer interpolation
     epicsType *pDataIn  = (epicsType *)pArray->pData;
     epicsType *pDataOut;
     NDArray *pArrayOut=NULL;
     size_t imageSize, rowSize, numRows;
     size_t dims[3];
     NDDimension_t tmpDim;
+    enum Colour {red, green, blue};
```
Comment (arvinderpalaha) [inline ADApp/pluginSrc/NDPluginColorConvert.cpp]: This should be changed to Color to keep spelling consistent

**Pull request merged**